stackit/internal/services/ske/cluster/resource.go

```diff
@@ -262,13 +266,35 @@ func (r *clusterResource) Schema(_ context.Context, _ resource.SchemaRequest, re
 			"name": schema.StringAttribute{
 				Description: "The cluster name.",
 				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
 				Validators: []validator.String{
 					validate.NoSeparator(),
 				},
 			},
+			"kubernetes_version_min": schema.StringAttribute{
+				Description: "The minimum Kubernetes version. This field will be used to set the kubernetes version on creation/update of the cluster. " + SKEUpdateDoc + " To get the current kubernetes version being used for your cluster, use the read-only `kubernetes_version_used` field. If unset, the latest supported Kubernetes version will be used.",
```

Comment:
I think this description and attribute name is problematic if we allow the user to downgrade versions. Maybe we should call this attribute "kubernetes_version_initial" or something similar, since it isn't really the minimum version if a user can downgrade a version after creation

Reply:
The user cannot downgrade a k8s version of a cluster after creation. This is validated with the `RequiresReplaceIf` function

Reply:
But can't a user specify a lower version which requires replace? Then the cluster is replaced and the version is updated, right?

My point is that with the attribute name of `kubernetes_version_min` the user may expect that lowering the minimum version won't change anything, whereas it might even replace the whole cluster if it is a downgrade.

Reply:
"kubernetes_version_initial" it's also not accurate because the version can be updated after the cluster is created. 

I got your point but I don't have an idea for a better name. Not being able to downgrade a kubernetes version is a common concept of SKE in general, but I've made it more explicit in the field description that a downgrade will require a replace of the cluster. I think that should be sufficient and clear for the users